Traceback (most recent call last):
  File "/opt/conda/envs/science/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/conda/envs/science/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/conda/envs/science/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/conda/envs/science/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/opt/conda/envs/science/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/opt/conda/envs/science/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/conda/envs/science/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
xmax = x.max() / xw
xmin = x.min() / xw
xspan = xmax - xmin
xlim = xmin - .1 * xspan, xmax +.1 * xspan

fig = plt.figure(figsize = (10, 6))
ax1 = fig.add_subplot(1,2,1)
plt.title("Whole solution")
plt.plot(t/Td, x/xw, "r-", label = "Whole solution")
plt.plot(t[-5*NF:]/Td, x[-5*NF:]/xw, "b-", label = "Last 5 periods")
plt.grid()
plt.legend()
plt.xlabel("Time, $t/T_d$")
plt.ylabel("Position $x/x_{w}$")
plt.ylim(*xlim)
ax2 = fig.add_subplot(1,2,2)
plt.title("Steady state")
plt.plot(t[-5*NF:]/Td, x[-5*NF:]/xw, "b-", label = "Last 5 periods")
plt.grid()
plt.legend()
plt.ylim(*xlim)
plt.xlabel("Time, $t/T_d$")
#plt.ylabel("Position $x/x_{w}$")
plt.show()
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[5], line 1[0m
[0;32m----> 1[0m xmax [38;5;241m=[39m [43mx[49m[38;5;241m.[39mmax() [38;5;241m/[39m xw
[1;32m      2[0m xmin [38;5;241m=[39m x[38;5;241m.[39mmin() [38;5;241m/[39m xw
[1;32m      3[0m xspan [38;5;241m=[39m xmax [38;5;241m-[39m xmin

[0;31mNameError[0m: name 'x' is not defined

